modvendor/modvendor_test.go

```diff
@@ -15,18 +15,843 @@
+		Error:    r.Error,
+	}
+	for _, vendored := range r.Vendored {
+		rawSource := fixupString(t, vendored, value)
+		modsrc, err := tf.ParseSource(rawSource)
+		assert.NoError(t, err)
+		out.Vendored[rawSource] = modvendor.Vendored{
+			Source: modsrc,
+			Dir:    modvendor.Dir(vendordir, modsrc),
+		}
+	}
+	for _, ignored := range r.Ignored {
+		rawSource := fixupString(t, ignored.RawSource, value)
+		reason := fixupString(t, ignored.ReasonPattern, value)
+		out.Ignored = append(out.Ignored, modvendor.IgnoredVendor{
+			RawSource: rawSource,
+			Reason:    reason,
+		})
+	}
+	return out
+}
+
+func evaluateWantedFiles(
+	t *testing.T,
+	wantFiles map[vendorPathSpec]fmt.Stringer,
+	modulesDir string,
+	rootdir string,
+	vendordir string,
+) map[string]fmt.Stringer {
+	evaluated := map[string]fmt.Stringer{}
+	for pathSpec, expectedStringer := range wantFiles {
+		pathSpecParts := strings.Split(string(pathSpec), "#")
+		assert.EqualInts(t, len(pathSpecParts), 2)
+		source := pathSpecParts[0]
+		path := pathSpecParts[1]
+
+		modsrc, err := tf.ParseSource(fixupString(t, source, modulesDir))
+		assert.NoError(t, err)
+		absVendorDir := modvendor.AbsVendorDir(rootdir, vendordir, modsrc)
+		evaluatedPath := filepath.Join(absVendorDir, path)
+		evaluated[evaluatedPath] = expectedStringer
+	}
+	return evaluated
+}
+
+func checkWantedFiles(
+	t *testing.T,
+	tc testcase,
+	modulesDir string,
+	rootdir string,
+	vendordir string,
+) {
+	wantFiles := evaluateWantedFiles(t, tc.wantFiles, modulesDir, rootdir, vendordir)
+	vendorDir := filepath.Join(rootdir, tc.vendordir)
+
+	if _, err := os.Stat(vendorDir); err != nil {
+		if os.IsNotExist(err) {
+			return
+		}
+		assert.Error(t, err)
+	}
+
+	err := filepath.Walk(vendorDir, func(path string, _ fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if filepath.Ext(path) != ".tf" {
+			return nil
+		}
+
+		expectedStringTemplate, ok := wantFiles[path]
+		if ok {
+			// file must be rewritten
+			relVendoredPaths := computeRelativePaths(
+				t, filepath.Dir(path), tc.wantVendored, modulesDir, rootdir, tc.vendordir,
+			)
+			want := fixupString(t, expectedStringTemplate.String(), relVendoredPaths)
+			got := string(test.ReadFile(t, filepath.Dir(path), filepath.Base(path)))
+			assert.EqualStrings(t, want, got, "file %q mismatch", path)
+		} else {
+			// check the vendored file is the same as the one in the module dir.
+			originalPath := strings.TrimPrefix(path, vendorDir)
+			pathEnd := strings.TrimPrefix(originalPath, modulesDir)
+			originalPath = strings.TrimSuffix(originalPath, pathEnd)
+			pathParts := strings.Split(pathEnd, "/")
+			moduleName := pathParts[1]
+
+			originalPath = filepath.Join(originalPath, moduleName, strings.Join(pathParts[3:], "/"))
+
+			originalBytes, err := ioutil.ReadFile(originalPath)
+			assert.NoError(t, err)
+
+			gotBytes, err := ioutil.ReadFile(path)
+			assert.NoError(t, err)
+			assert.EqualStrings(t, string(originalBytes), string(gotBytes),
```

Comment:
hmm maybe using test.AssertDiff would produce a better output for comparing multiline strings